datalad/metadata/search.py

```diff
@@ -124,6 +124,8 @@ def __call__(match,
                   match, dataset)
         try:
             ds = require_dataset(dataset, check_installed=True, purpose='dataset search')
+            if ds.id is None:
+                raise NoDatasetArgumentFound
```

Comment:
Why? We decided to care about datasets and not repositories. Otherwise I should also acknowledge the existence of a directory. IMHO an appropriate message is generated few lines below. This particular exception never sees the light of day anyway.